### PR TITLE
Use SQL expression based config by default

### PR DESCRIPTION
**Pull request by jaymo001**

### CODE REVIEW

File: feathr_project/test/test_feature_anchor.py [hunk 1]
```diff
@@ -140,12 +140,12 @@ def test_agg_anchor_to_config():
...
             aggregationFeatures: {
                 source: nycTaxiBatchSource
-                key: [DOLocationID]
+                key.sqlExpr: [DOLocationID]
                 features: {
                     f_location_avg_fare: {
                         def: "cast_float(fare_amount)"
                         window: 90d
-                        agg: AVG
+                        aggregation: AVG
```
Comment (d4ve) [inline feathr_project/test/test_feature_anchor.py:148]: What does this change do?

File: feathr_project/feathr/anchor.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def to_feature_config(self) -> str:
         tm = Template("""
             {{anchor_name}}: {
                 source: {{source.name}}
-                key: [{{key_list}}]
+                key.sqlExpr: [{{key_list}}]
```
Comment (d4ve) [inline feathr_project/feathr/anchor.py:37]: In general this makes sense, but is there any effect on compatibility? Is the change motivated by performance?

Comment (jaymo001) [reply feathr_project/feathr/anchor.py:37]: No. It is compatible. Yes, It is motivated by performance.

File: feathr_project/test/test_feature_anchor.py (hunk 1 above)
Comment (jaymo001) [reply feathr_project/test/test_feature_anchor.py:148]: It should be aggregation instead of agg.

**Review by d4ve — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**